limits_tools/limits_tools.py

```diff
@@ -0,0 +1,130 @@
+from math import sqrt, pi
+
+from ROOT import TGraphAsymmErrors
+
+import physics as ph
+
+regions_to_mask = {
+    "rho/omega": (0.74, 0.82),
+    "phi": (0.97, 1.07),
+    "j/psi": (2.94, 3.24),
+    "psi": (3.50, 3.86),
+    "z": (86.63, 95.75),
+}
+
+
+# with all couplings on
+alp_cross_section_all_couplings_on = {
+# mass (GeV), x_sec (pb)
+    0.1:  3.102,
+    0.2: 3.066,
+    0.3: 3.075,
+    0.315: 3.122,
+    0.5: 3.098,
+    1.0: 3.104,
+    2.0: 3.087,
+    4.0: 3.057,
+    8.0: 3.023,
+    8.5: 3.086,
+    10.: 3.046,
+    20.: 2.993,
+    40.: 2.870,
+    50.: 2.799,
+    70.: 2.622,
+    80.: 2.518,
+    # 90.: 2.424,
+}
+
+# with only alp-top couplings on (cu = -0.5, cq = 0.5)
+alp_cross_section_only_top_coupling = {
+# mass (GeV), x_sec (pb)
+    0.1:  0.1188,
+    1.0: 0.1169,
+    10.: 0.1148,
+    40.: 0.09168,
+    50.: 0.0844,
+    80.: 0.06463,
+}
+
+
+def find_lifetime_for_mass(mass, coupling, Lambda):
+    ctau = ph.ctaua(mass, coupling, coupling, Lambda)  # in cm
+    
+    boost = 1 / mass
+    
+    if mass < 3:
+        boost *= 223
+    elif mass < 20:
+        boost *= 230
+    elif mass < 30:
+        boost *= 240
+    elif mass < 50:
+        boost *= 260
+    else:
+        boost *= 296
+    
+    return boost * ctau
+
+
+def mass_to_lifetime(input_graph, coupling, Lambda):
+    output_graph = TGraphAsymmErrors()
+    
+    for i in range(input_graph.GetN()):
+        mass = input_graph.GetPointX(i)
+        x_sec = input_graph.GetPointY(i)
+        x_sec_up = input_graph.GetErrorYhigh(i)
+        x_sec_down = input_graph.GetErrorYlow(i)
+        
+        lifetime = find_lifetime_for_mass(mass, coupling, Lambda)
+        
+        output_graph.SetPoint(i, lifetime, x_sec)
+        output_graph.SetPointError(i, 0, 0, x_sec_down, x_sec_up)
+    
+    return output_graph
+
+
+def find_coupling_for_cross_section(x_sec, mass):
+    coeffs = {
+        0.1: 0.1177,
```

Comment:
What are these coefficient values? Are they also cross sections?

Reply:
This are coefficients in translating coupling to cross section: σ = coeff * c_tt^2.